docs: rearrange docs structure and prepare for future outline (#2051)
namehash/ensnode: docs/ensnode.io/src/content/docs/docs/components/index.mdx

---

> title: ENSNode components

Reviewer: ```suggestion
title: ENSNode services
```

> description: Microservices that make up ENSNode—ENSApi, ENSIndexer, ENSDb, ENSRainbow, and ENSAdmin—and how to work on each.

Reviewer: ```suggestion
description: Introduction to the services that make up the ENSNode stack, including ENSApi, ENSIndexer, ENSDb, ENSRainbow, and ENSAdmin.
```

sidebar:
  label: Overview
  order: 1
---

import { LinkCard, CardGrid } from "@astrojs/starlight/components";

**Developing or contributing to ENSNode?** Use these sections for concepts, deployment, configuration, and contribution guides for the part of the stack you care about.

> ENSNode is built from **microservices**. Each service has a clear role in indexing, storage, APIs, healing, or operations.

Reviewer: ```suggestion
Each ENSNode instance is composed of **a set of services**. Each service has a clear role in indexing, storing the onchain state of ENS, APIs, name healing, or operational admin.
```

Goal: For some, the word "microservices" is a red flag. Let's call them "services", not "microservices"

<CardGrid>

>   <LinkCard title="ENSApi" href="/docs/components/ensapi" description="GraphQL, REST, and operational configuration for the API service." />

Reviewer: ```suggestion
  <LinkCard title="ENSApi" href="/docs/components/ensapi" description="APIs for building on ENS." />
```

>   <LinkCard title="ENSIndexer" href="/docs/components/ensindexer" description="Blockchain indexing with Ponder." />

Reviewer: ```suggestion
  <LinkCard title="ENSIndexer" href="/docs/components/ensindexer" description="Multichain indexer for ENSv2." />
```

>   <LinkCard title="ENSDb" href="/docs/components/ensdb" description="PostgreSQL schema and direct data access." />

Reviewer: ```suggestion
  <LinkCard title="ENSDb" href="/docs/components/ensdb" description="The live, onchain state of ENS in your DB." />
```

>   <LinkCard title="ENSRainbow" href="/docs/components/ensrainbow" description="Label healing and rainbow tables." />

Reviewer: ```suggestion
  <LinkCard title="ENSRainbow" href="/docs/components/ensrainbow" description="Heals unknown ENS names." />
```

>   <LinkCard title="ENSAdmin" href="/docs/components/ensadmin" description="Operator dashboard for indexed ENS state." />

Reviewer: ```suggestion
  <LinkCard title="ENSAdmin" href="/docs/components/ensadmin" description="ENSNode operator dashboard and ENS Protocol Inspector." />
```
